pkg/virt-controller/watch/vmi.go

```diff
@@ -380,6 +380,40 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 			conditionManager.RemoveCondition(vmiCopy, virtv1.VirtualMachineInstanceConditionType(k8sv1.PodReady))
 		}
 
+		var networkPatchNeeded bool
+		existingNetworks := map[string]virtv1.Network{}
+		for _, network := range vmiCopy.Spec.Networks {
+			if network.Name != "" {
+				existingNetworks[network.Name] = network
+			}
+		}
+		for i, iface := range vmiCopy.Status.Interfaces {
+			if existingNetworks[iface.Name].NetworkSource.Pod != nil && vmiCopy.Status.Interfaces[i].IP != pod.Status.PodIP {
+				vmiCopy.Status.Interfaces[i].IP = pod.Status.PodIP
+				networkPatchNeeded = true
+
+				for j, ip := range iface.IPs {
+					if ip == iface.IP {
+						vmiCopy.Status.Interfaces[i].IPs[j] = pod.Status.PodIP
+					}
+				}
+				// Currently will only traverse to the first Pod network found
+				break
+			}
+		}
+		if networkPatchNeeded {
```

Comment:
On line 417 and 431 we have some comments regarding the patching. Would it make sense to have them here as well?

Reply:
Actually, this can be changed to use the new `patchOps` and Conditions checks on latest code.
Changed it now.